appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -1471,6 +1471,14 @@ Blockly.ReplMgr.getFromRendezvous = function() {
                 rs.versionurl = 'http://' + json.ipaddr + ':8001/_getversion';
                 rs.baseurl = 'http://' + json.ipaddr + ':8001/';
                 rs.android = !(new RegExp('^i(pad)?os$').test((json.os || 'Android').toLowerCase()));
+                if (!(rs.android) && Blockly.ReplMgr.hasExtensions()){
+                    var ios_dialog = new Blockly.Util.Dialog(Blockly.Msg.EXTENSIONS, Blockly.Msg.EXTENSIONS_iOS, Blockly.Msg.REPL_CANCEL, true, null, 0, function(){
+                        ios_dialog.hide();
+                        top.ReplState.state = Blockly.ReplMgr.rsState.IDLE;
+                        top.BlocklyPanel_indicateDisconnect();
+                        rs.connection = null;
```

Comment:
I think we want to move these into the outer scope after the dialog is made and then return. Right now as implemented the system will keep trying to continue the process, and how far it gets will depend on how long it takes for someone to close the dialog. Instead, we probably want to cancel the connection immediately and return as soon as the dialog is shown (the dialog doesn't block).

Reply:
Oh ok makes sense. If the only thing in the callback is dialog.hide() can that be put as a direct argument the the new dialog or do I still have to define the anonymous function with it inside?
e.g ios_dialog = new Blockly.Util.Dialog(Blockly.Msg.EXTENSIONS, Blockly.Msg.EXTENSIONS_iOS, Blockly.Msg.REPL_CANCEL, true, null, 0, ios_dialog.hide()) ? I would assume define a function but I'd rather ask just in case